### DISCUSSION TITLE
KAMP and BED LEVELING toggles

### DISCUSSION THREAD

**Discussion by jonny7737** [Guilouz/Creality-Helper-Script-Wiki#541] [General]
What is the proper use of these two toggles. KAMP seems obvious and BED LEVELING seems obvious but why the two toggles?  Isn't KAMP just targeted bed leveling?
What happens if both are enabled?
What happens when only one is enabled?

Just something I think about when I look at the UI.

**Comment by mattskent** [comment]
Pretty sure the documentation (maybe the changelog?) goes over this. If both are enabled or if just KAMP is enabled, it will use KAMP. If only bed leveling is enabled, it will use the default calibration where it does a whole bed mesh. If neither are enabled, it won't do any new leveling for that print.

**Comment by jonny7737** [comment]
Changelog. I should have thought of that.

Improved START_PRINT macro, you can now disable KAMP with KAMP button on Web interfaces.
When KAMP is enabled: A bed mesh named "kamp" is created and loaded to use adaptative bed mesh.
When KAMP is disabled: A bed mesh named "default" is created and loaded to use classic calibration (full bed mesh), you can also disable bed leveling before print with BED_LEVELING button on Web interfaces (it have no effect when KAMP is enabled).

Thanks.

**Comment by Guilouz** [comment]
https://guilouz.github.io/Creality-Helper-Script-Wiki/helper-script/klipper-adaptive-meshing-and-purging/#use
